client/tests/integration/smartcontracts/validator_with_admin/src/lib.rs

```diff
@@ -116,34 +122,39 @@ impl ExpressionEvaluator for CustomValidator {
 }
 
 #[entrypoint]
-pub fn migrate() -> MigrationResult {
-    CustomValidator::migrate()
+pub fn migrate(block_height: u64) -> MigrationResult {
+    CustomValidator::migrate(block_height)
```

Comment:
This seems like a costly oversight. We should have a model for the upgrade mechanism

Reply:
I don't get it. Can you explain more, please?

Reply:
Yes, previously we could do that using inderect proofs. Right now we have `block_height` and `DefaultValidator` will fail to migrate if it was uploaded not as the first block